src/drive/web/modules/views/Folder/FolderViewBody.jsx

```diff
@@ -21,6 +21,32 @@ import SelectionBar from 'drive/web/modules/selection/SelectionBar'
 import { TRASH_DIR_ID } from 'drive/constants/config'
 import { openLocalFile } from 'drive/mobile/modules/offline/duck'
 
+const generateFileUrl = ({ file, client, isFlatDomain }) => {
+  if (file.class === 'shortcut') {
+    const currentURL = new URL(window.location)
+    let webLink = ''
+    if (currentURL.pathname === '/public') {
+      webLink = generateWebLink({
+        cozyUrl: client.getStackClient().uri,
+        pathname: '/public',
+        slug: 'drive',
+        hash: `external/${file.id}`,
+        searchParams: currentURL.searchParams,
+        subDomainType: isFlatDomain ? 'flat' : 'nested'
+      })
+    } else {
+      webLink = generateWebLink({
+        cozyUrl: client.getStackClient().uri,
+        pathname: '/',
+        slug: 'drive',
+        hash: `external/${file.id}`,
+        subDomainType: isFlatDomain ? 'flat' : 'nested'
+      })
+    }
+    return webLink
+  }
+}
```

Comment:
Why did we move this method in this file?

Reply:
I think it's better to have all the file opening logic in the same place, rather than split between FileExplorer and File as it was before. For example [this comment](https://github.com/cozy/cozy-drive/blob/v1.25.0/src/drive/web/modules/navigation/FileExplorer.jsx#L193) was related to the [`if !isMobile` in FIle.jsx](https://github.com/cozy/cozy-drive/blob/v1.25.0/src/drive/web/modules/filelist/FileOpener.jsx#L25), but with the 2 being very far away it was hard to understand.

Reply:
You totally right about the opening stuff. My concern was more about the `generateFileUrl` method. I'm more and more convinced that JS only stuff should not be in a JSX file. I'm sure that `generateFileUrl` can be used outside of this component

Reply:
Oh ok! Yep no problem I'll move it in a standalone file.